test/fakeintake/server/server.go

```diff
@@ -79,18 +76,28 @@ func WithPort(port int) func(*Server) {
 func WithReadyChannel(ready chan bool) func(*Server) {
 	return func(fi *Server) {
 		if fi.URL() != "" {
-			fmt.Printf("Fake intake is already running. Stop it and try again to change the ready channel.")
+			log.Println("Fake intake is already running. Stop it and try again to change the ready channel.")
 			return
 		}
 		fi.ready = ready
 	}
 }
 
+func WithClock(clock clock.Clock) func(*Server) {
+	return func(fi *Server) {
+		if fi.URL() != "" {
```

Comment:
I added the check as I considered that, since `With` functions are public, they can potentially be called out of `NewServer`. Is there a way to prevent this ?

```golang
import 'fakeintake/server' as 'fakeintake'

srv := fi.NewServer()
fi.WithClock(clk)(srv)
```